app/javascript/App/bundles/ExchangeListings/client.js

```diff
@@ -6,22 +6,141 @@ import LayoutMobile from '../../components/LayoutMobile'
 import ListingsHeader from './components/ListingsHeader'
 import ListingsList from './components/ListingsList'
 import BodySection from './components/BodySection'
+import localAPI from '../../lib/localAPI'
 
 class ExchangeListingsPage extends Component {
-  state = { showFilterPanel: false }
+  constructor(props) {
+    super(props)
 
-  toggleFilterPanel() {
+    this.state = {
+      listings: props.initialListings,
+      newestDetectedAt: props.initialListings[0].detected_at,
+      oldestDetectedAt:
+        props.initialListings[props.initialListings.length - 1].detected_at,
+      hasMore: true,
+      showFilterPanel: false,
+      selectedSymbols: [],
+      exchangeSlugs: [],
+      detectedSince: null,
+      detectedUntil: null,
+    }
+  }
+
+  componentDidMount() {
+    window.addEventListener('resize', debounce(() => this.forceUpdate(), 500))
+  }
+
+  componentDidUpdate() {
+    // TODO: Get this working!
```

Comment:
New ticket spun up for this:
https://app.asana.com/0/683768070860536/791217413395046